### ISSUE TITLE
blazor.boot.json not found

### ISSUE THREAD

**Issue by anders130** [FlorianRappl/Piral.Blazor#126]
# Bug Report
## Environment Details and Version

[Package Version, Blazor Version, OS, Browser,...]
- Piral Version: `1.3.3`
- Blazor Verion: `net7.0`
- OS: `Windows 11`
- Browser: `Firefox`, `Edge`, `Chrome`
- Node Version: `v20.7.0`

## Description

I was following the [tutorial on how to create a Blazor Pilet from the Visual Studio Template](https://www.youtube.com/watch?v=Ychzp2xMxes). For the **PiralInstance** i was using both `sample-piral` and a custom one following the [tutorial on the homepage](https://docs.piral.io/guidelines/tutorials/02-getting-started) (without blazor, that instance is working).
When trying to start the blazor Pilet from Visual Studio, I get the following errors:

```txt
Starting Piral.Blazor.DevServer ...

  applicationPath = path\to\project\BlazorApp1\BlazorApp1\bin\Debug\net7.0\BlazorApp1.dll
  forwardedPaths =
  remoteFeedUrl =
  piralInstance = my-app
  piletDir = path\to\project\BlazorApp1\BlazorApp1\..\piral~\BlazorApp1
  outPath = ..\piral~
  appId = BlazorApp1
  feed = http://localhost:51660

[piral-cli] + piral-cli v1.3.3
[piral-cli] - Reading configuration ...
[piral-cli] [watch] build finished, watching for changes...
[piral-cli] i > Running at http://localhost:51660/
[piral-cli] i > Manage via http://localhost:51660/manage-mock-server
[piral-cli] X [ERROR] Could not find the "blazor.boot.json" in staticwebassets.build.json. Something seems to be wrong. [plugin codegen-loader]
[piral-cli]
[piral-cli]     src/blazor.codegen:1923:10:
[piral-cli]       1923 Ôöé     throw new Error(`Could not find the "${bbjson}" in ${swajson...
[piral-cli]            ÔòÁ           ^
[piral-cli]
[piral-cli]     at prepare (path\to\project\BlazorApp1\piral~\BlazorApp1\src\blazor.codegen:1923:11)
[piral-cli]     at module.exports (path\to\project\BlazorApp1\piral~\BlazorApp1\src\blazor.codegen:2205:78)
[piral-cli]     at async path\to\project\BlazorApp1\piral~\BlazorApp1\node_modules\esbuild-codegen-plugin\lib\index.js:20:30
[piral-cli]     at async requestCallbacks.on-load (path\to\project\BlazorApp1\piral~\BlazorApp1\node_modules\piral-cli-esbuild\node_modules\esbuild\lib\main.js:1434:22)
[piral-cli]     at async handleRequest (path\to\project\BlazorApp1\piral~\BlazorApp1\node_modules\piral-cli-esbuild\node_modules\esbuild\lib\main.js:729:13)
[piral-cli]
[piral-cli]   This error came from the "onLoad" callback registered here:
[piral-cli]
[piral-cli]     node_modules/esbuild-codegen-plugin/lib/index.js:16:14:
[piral-cli]       16 Ôöé         build.onLoad({ filter: /\.codegen$/ }, async (args) => {
[piral-cli]          ÔòÁ               ~~~~~~
[piral-cli]
[piral-cli]     at setup (path\to\project\BlazorApp1\piral~\BlazorApp1\node_modules\esbuild-codegen-plugin\lib\index.js:16:15)
[piral-cli]     at handlePlugins (path\to\project\BlazorApp1\piral~\BlazorApp1\node_modules\piral-cli-esbuild\node_modules\esbuild\lib\main.js:1292:21)
[piral-cli]     at buildOrContextImpl (path\to\project\BlazorApp1\piral~\BlazorApp1\node_modules\piral-cli-esbuild\node_modules\esbuild\lib\main.js:978:5)
[piral-cli]     at Object.buildOrContext (path\to\project\BlazorApp1\piral~\BlazorApp1\node_modules\piral-cli-esbuild\node_modules\esbuild\lib\main.js:786:5)
[piral-cli]     at path\to\project\BlazorApp1\piral~\BlazorApp1\node_modules\piral-cli-esbuild\node_modules\esbuild\lib\main.js:2186:68
[piral-cli]     at new Promise (<anonymous>)
[piral-cli]     at Object.context (path\to\project\BlazorApp1\piral~\BlazorApp1\node_modules\piral-cli-esbuild\node_modules\esbuild\lib\main.js:2186:27)
[piral-cli]     at context (path\to\project\BlazorApp1\piral~\BlazorApp1\node_modules\piral-cli-esbuild\node_modules\esbuild\lib\main.js:2026:58)
[piral-cli]     at Object.bundle (path\to\project\BlazorApp1\piral~\BlazorApp1\node_modules\piral-cli-esbuild\lib\esbuild\bundler-run.js:52:53)
[piral-cli]
[piral-cli]   The plugin "codegen-loader" was triggered by this import
[piral-cli]
[piral-cli]     src/index.tsx:1:27:
[piral-cli]       1 Ôöé ´╗┐import * as Blazor from './blazor.codegen';
[piral-cli]         ÔòÁ                         ~~~~~~~~~~~~~~~~~~
[piral-cli]
[piral-cli] ÔêÜ Ready!
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: http://localhost:5148
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Development
info: Microsoft.Hosting.Lifetime[0]
      Content root path: path\to\project\BlazorApp1\BlazorApp1\bin\Debug\net7.0
[piral-cli] 1 error
info: System.Net.Http.HttpClient.Default.LogicalHandler[100]
      Start processing HTTP request GET http://localhost:51660/$pilet-api
info: System.Net.Http.HttpClient.Default.ClientHandler[100]
      Sending HTTP request GET http://localhost:51660/$pilet-api
    
```

## Steps to Reproduce

1. Adapt standard blazor wasm project template as described in the tutorial video.
2. Build the project
3. Start the project via Visual Studio

## Expected behavior

No exception while running the program.

## Actual behavior

App-shell without the blazor pilet is running

**Comment by FlorianRappl** [comment]
Can you provide the sln / csproj in a GitHub repo? There has to be something wrong in your setup if no blazor.boot.json is created as output file from the build process.

**Comment by anders130** [comment]
I have quickly moved the files together into this repo https://github.com/anders130/piral-blazor-demo.git.
The [my-app-1.0.0.tgz](https://github.com/anders130/piral-blazor-demo/blob/master/my-app-1.0.0.tgz) is just copied from the dist/emulator off my-app.

**Comment by FlorianRappl** [comment]
Thanks for the reproducible - I think this might be a change in the SDK as I can see that a previous "AssetTraitName" in the SWA being `BlazorWebAssemblyResource` is now `WasmResource` (not sure about the reason, all I can say is that this is new - might be related to the .NET 8 SDK Preview). This needs to be fixed.

**Comment by FlorianRappl** [comment]
Fixed / published as preview package (version: 7.0.10-pre.20231110.1)
